DependencyInjection/Compiler/CsrfExtensionPass.php

```diff
@@ -29,6 +29,17 @@ public function process(ContainerBuilder $container)
                 $tokenStorageReference = new Reference('security.context');
             }
             $definition->replaceArgument(0, $tokenStorageReference);
+
+            // Symfony <2.8 BC
+            $typeExtensionTag = $definition->getTag('form.type_extension');
+            if (method_exists($definition, 'setShared')) {
```

Comment:
The alias here must be the same thing that the return value of ``getExtendedType`` in the type extension (and Symfony 2.8 now validates it early to avoid hard-to-debug issues). So you need to update the type extension to be compatible with 2.8 without deprecations too (your current code is not compatible at all because of the mismatch)